src/components/VideoPlayer/BaseVideoPlayer.js

```diff
@@ -93,9 +101,14 @@ function BaseVideoPlayer({
             }
             videoResumeTryNumber.current -= 1;
         },
-        [playVideo],
+        [playVideo, videoResumeTryNumber],
     );
 
+    const updateIsPlaying = useCallback((isVideoPlaying) => {
+        setIsPlaying(isVideoPlaying);
+        isPlayingRef.current = isVideoPlaying;
+    }, []);
```

Comment:
why do we need both `isPlaying` as a state and `isPlaying` as a ref?

Reply:
It is connected to your comment below. For some reason, if we use a state variable like `isPlaying` inside `handleFullscreenUpdate` function, and we play the video in fullscreen mode, pausing the video will set `isPlaying` to false, so the function reference will change. This will dismiss the whole fullscreen mode. I'm not sure why it is happening, it is probably a bug in `expo-av` library. Using ref fixes that. 
I will try to change it so it won't look suspicious 😅

Reply:
Here is what happens if we use `isPlaying` state variable inside `handleFullscreenUpdate ` function

https://github.com/Expensify/App/assets/39538890/08e75045-21a2-40c4-a708-03eec4b21065

Reply:
What about `videoStateRef` instead of `isPlayingRef`?